scheduler/src/cook/kubernetes/api.clj

```diff
@@ -213,18 +213,25 @@
     ; We want to process all changes through the callback process.
     ; So compute the delta between the old and new and process those via the callbacks.
     ; Note as a side effect, the callbacks mutate all-pods-atom
-    (doseq [task (set/union new-pod-names old-pod-names)]
-      (log/info "In" compute-cluster-name "compute cluster, pod watch doing callback"
-                {:new? (contains? new-pod-names task)
-                 :old? (contains? old-pod-names task)
-                 :pod-name task})
-      (doseq [callback callbacks]
-        (try
-          (callback task (get old-all-pods task) (get namespaced-pod-name->pod task))
-          (catch Exception e
-            (log/error e "In" compute-cluster-name
-                       "compute cluster, pod watch error while processing callback for" task)))))
-
+    (let [tasks (set/union new-pod-names old-pod-names)
+          futures
+          (doall
+            (map (fn [task]
+                   (.submit
+                     launch-task-executor-service
```

Comment:
why re-use launch-task-executor-service and not something that's just for the parallel node watch? can there be contention?

Reply:
Also seems like we might want to configure a different number of threads for task-launching than for pod processing

Reply:
No. The choice to use the same executor says that we have one control for maximum concurrency level though tthe state machine. We want this. as it controls the maximum outstanding requests we'll make to the k8s server. In addition, both the scan at watch startup as well as the launch rate are bulk, high-throughput operations that occur intermittently. Now we have one maximum knob to set instead of the maximum load being the sum of both of these (and hard synchronize to loadtest)

Reply:
Can we give it a more appropriate name?

Reply:
(given that it's now going to do more than launching tasks)

Reply:
Will do. controller-num-threads (and renamed the executor service correspondingly)

Reply:
There can be contention in the threads, but that's planned contention. We want a knob to control max outstanding k8s requests, and the threadpool size is it.